### PR TITLE
Coordinator setup

### PR DESCRIPTION
**Pull request by GLTaylor**

### CODE REVIEW

File: CultivateMVP/Coordinator/SceneCoordinator.swift [hunk 1]
```diff
@@ -1,10 +1,87 @@
...
+    func transition(to scene: Scene, type: SceneTransitionType) -> Observable<Void>
+
+    @discardableResult
+    func pop(animated: Bool) -> Observable<Void>
+}
+
+struct SceneCoordinator: SceneCoordinatorType {
+
+    //I think I can put the first one for this in the app delegate
+    private let vCFactory: SceneFactory
```
Comment (AlexShubin) [inline CultivateMVP/Coordinator/SceneCoordinator.swift]: vcFactory?

File: CultivateMVP/JournalOptionsScene/JournalOptionsViewModel.swift [hunk 2]
```diff
@@ -1,9 +1,17 @@
...
-//
-
 import Foundation
+
+protocol JournalOptionsViewModelType {
+    var entries: [Entry] { get }
+    var moodEntry: MoodEntry { get }
+}
+
+class JournalOptionsViewModel: JournalOptionsViewModelType {
```
Comment (AlexShubin) [inline CultivateMVP/JournalOptionsScene/JournalOptionsViewModel.swift]: VMs can be also structs :)

File: CultivateMVP/Coordinator/Scene.swift [hunk 3]
```diff
@@ -0,0 +1,38 @@
+import UIKit
+
+enum Scene {
+    case thoughtOfDay(ThoughtOfDayViewModel)
+    case journalOptions(JournalOptionsViewModel)
+    case finished // Todo
+}
+
+class SceneFactory {
```
Comment (AlexShubin) [inline CultivateMVP/Coordinator/Scene.swift]: Also can be a struct...

File: CultivateMVP/Coordinator/SceneCoordinator.swift (hunk 1 above)
Comment (GLTaylor) [reply CultivateMVP/Coordinator/SceneCoordinator.swift]: whoops

**Review by AlexShubin — APPROVED** [review]

**Pull request merged**